activation/post_supervisor.go

```diff
@@ -232,67 +225,72 @@ func (ps *PostSupervisor) runCmd(
+		"--labels-per-unit", strconv.FormatUint(uint64(postCfg.LabelsPerUnit), 10),
+		"--k1", strconv.FormatUint(uint64(postCfg.K1), 10),
+		"--k2", strconv.FormatUint(uint64(postCfg.K2), 10),
+		"--k3", strconv.FormatUint(uint64(postCfg.K3), 10),
+		"--pow-difficulty", postCfg.PowDifficulty.String(),
+
+		"--dir", postOpts.DataDir,
+		"-n", strconv.FormatUint(uint64(postOpts.Scrypt.N), 10),
+		"-r", strconv.FormatUint(uint64(postOpts.Scrypt.R), 10),
+		"-p", strconv.FormatUint(uint64(postOpts.Scrypt.P), 10),
+
+		"--threads", strconv.FormatUint(uint64(provingOpts.Threads), 10),
+		"--nonces", strconv.FormatUint(uint64(provingOpts.Nonces), 10),
+		"--randomx-mode", provingOpts.RandomXMode.String(),
+
+		"--watch-pid", strconv.Itoa(os.Getpid()),
+	}
+	if cmdCfg.MaxRetries > 0 {
+		args = append(args, "--max-retries", strconv.Itoa(cmdCfg.MaxRetries))
+	}
+	if cmdCfg.CACert != "" {
+		args = append(args, "--ca-cert", cmdCfg.CACert)
+	}
+	if cmdCfg.Cert != "" {
+		args = append(args, "--cert", cmdCfg.Cert)
+	}
+	if cmdCfg.Key != "" {
+		args = append(args, "--key", cmdCfg.Key)
+	}
 
-		cmd := exec.CommandContext(
-			ctx,
-			cmdCfg.PostServiceCmd,
-			args...,
-		)
-		cmd.Dir = filepath.Dir(cmdCfg.PostServiceCmd)
-		pipe, err := cmd.StderrPipe()
-		if err != nil {
-			ps.logger.Error("setup stderr pipe for post service", zap.Error(err))
-			return nil
-		}
+	cmd := exec.CommandContext(
+		ctx,
+		cmdCfg.PostServiceCmd,
+		args...,
+	)
+	cmd.Dir = filepath.Dir(cmdCfg.PostServiceCmd)
+	pipe, err := cmd.StderrPipe()
+	if err != nil {
+		ps.logger.Error("setup stderr pipe for post service", zap.Error(err))
+		return nil
+	}
 
-		var eg errgroup.Group
-		eg.Go(ps.captureCmdOutput(pipe))
-		if err := cmd.Start(); err != nil {
-			pipe.Close()
-			ps.logger.Error("start post service", zap.Error(err))
-			return nil
-		}
-		ps.logger.Info("post service started", zap.Int("pid", cmd.Process.Pid), zap.String("cmd", cmd.String()))
-		ps.pid.Store(int64(cmd.Process.Pid))
-		events.EmitPostServiceStarted()
-		err = cmd.Wait()
-		if err := ctx.Err(); err != nil {
-			events.EmitPostServiceStopped()
-			if err := eg.Wait(); err != nil {
-				ps.logger.Warn("output reading goroutine failed", zap.Error(err))
-			}
-			return nil
+	var eg errgroup.Group
+	eg.Go(ps.captureCmdOutput(pipe))
+	if err := cmd.Start(); err != nil {
+		pipe.Close()
+		ps.logger.Error("start post service", zap.Error(err))
+		return nil
+	}
+	ps.logger.Info("post service started", zap.Int("pid", cmd.Process.Pid), zap.String("cmd", cmd.String()))
+	ps.pid.Store(int64(cmd.Process.Pid))
+	events.EmitPostServiceStarted()
+	err = cmd.Wait()
+	if ctx.Err() != nil {
+		events.EmitPostServiceStopped()
+		if err := eg.Wait(); err != nil {
+			ps.logger.Warn("output reading goroutine failed", zap.Error(err))
 		}
-		ps.logger.Warn("post service exited", zap.Error(err))
-		eg.Wait()
+		return nil
 	}
+	eg.Wait()
+	ps.logger.Fatal("post service exited", zap.Error(err))
```

Comment:
i think it makes more sense to raise Fatal in node when error will bubble up. there are several examples of this, for example peersync process

Reply:
You mean just returning an error here and handle the error with `log.Fatal` in `node.go`? 

The problem is that the error here happens asynchronously and doesn't bubble up at the moment. Returning it here would do nothing (besides stopping the internal `errgroup` and preventing it from accepting new go routines).